src/views/document/utils/DocumentViewMixin.js

```diff
@@ -144,11 +144,34 @@ export default {
                     this.$set(this.promise, 'data', this.draft)
                 })
             } else { // normal mode
-                if (this.document && $route.params.id === this.document.document_id && this.expected_lang === this.lang) {
+                // because of updateUrl(), we may have nothing to do
+                if (this.document && parseInt($route.params.id, 10) === this.document.document_id && this.expected_lang === this.lang) {
                     return
                 }
 
-                this.promise = c2c[this.documentType].getCooked(this.documentId, this.expected_lang).then(this.updateUrl)
+                this.promise = c2c[this.documentType].getCooked(this.documentId, this.expected_lang)
+                    .then(this.scrollToHash)
+                    .then(this.updateUrl)
+            }
+        },
+
+        scrollToHash() {
+            if (this.$route.hash) {
+                // we'll have to wait for DOM update
+                this.$nextTick(() => {
+                    let el = document.querySelector(this.$route.hash)
+
+                    if (el) {
+                        let docEl = document.documentElement
+                        let docRect = docEl.getBoundingClientRect()
+                        let elRect = el.getBoundingClientRect()
+                        let x = elRect.left - docRect.left
+                        let y = elRect.top - docRect.top
+                        window.scrollTo(x, y - 50) // navbar height ...
```

Comment:
I don't understand why you are computing an `x` position: you're only trying to sroll vertically to the right anchor position?

Reply:
sure :)